src/dbnode/client/types.go

```diff
@@ -594,6 +594,12 @@ type Options interface {
 
 	// NewConnectionFn returns the new connection generator function.
 	NewConnectionFn() NewConnectionFn
+
+	// SetNamespaceInitializer sets the NamespaceInitializer for watching namespace definition changes.
```

Comment:
nit: Should this be a `Watcher` rather than an initializer?

Reply:
Technically, this initializer generates a `Registry` object that you can then create watches with. I think it probably makes more sense to make that clearer in the comment.